### PR TITLE
Add stream context providers

### PR DESCRIPTION
**Pull request by s-vitaliy**
## Scope

Implemented:
- Added the `IStreamContext` interface for stream implementations.
- Added the `AddStreamContext` extension method fir `IServiceCollection`

## Checklist

- [ ] GitHub issue exists for this change.
- [ ] Unit tests added and they pass.
- [ ] Line Coverage is at least 80%.
- [x] Review requested on `latest` commit.

### CODE REVIEW

File: src/Providers/StreamContext/StreamContextProvider.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    /// <summary>
+    /// Stream context provider that uses the Environment variables for populating IStreamContext properties
+    /// </summary>
+    /// <typeparam name="TStreamContext">Stream context type</typeparam>
+    /// <returns>Stream context instance</returns>
+    public static TStreamContext ProvideFromEnvironment<TStreamContext>() where TStreamContext : class, IStreamContextWriter, IStreamContext
+    {
+        var context = JsonSerializer.Deserialize<TStreamContext>(EnvironmentExtensions.GetAssemblyEnvironmentVariable("SPEC"));
+        context.SetIsRunningInBackfillMode(EnvironmentExtensions.GetAssemblyEnvironmentVariable("FULL_LOAD")
```
Comment (george-zubrienko) [inline src/Providers/StreamContext/StreamContextProvider.cs:39]: call this BACKFILL now? :)

File: src/Services/Base/IStreamContext.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+{
+    /// <summary>
+    /// Id of the stream
+    /// </summary>
+    string StreamId { get; }
+
+    /// <summary>
+    /// True if stream is running in backfill (full reload) mode
+    /// </summary>
+    bool IsRunningInBackfillMode { get; }
```
Comment (george-zubrienko) [inline src/Services/Base/IStreamContext.cs:16]:
```suggestion
    bool IsBackfilling { get; }
```

**Review by george-zubrienko — APPROVED** [review]
tiny naming suggestions

File: src/Providers/StreamContext/StreamContextProvider.cs (hunk 1 above)
Comment (s-vitaliy) [reply src/Providers/StreamContext/StreamContextProvider.cs]: done

**Pull request merged**